packages/contracts/contracts/interfaces/Interpreter.sol

```diff
@@ -0,0 +1,15 @@
+pragma solidity 0.5.8;
+pragma experimental "ABIEncoderV2";
+
+
+contract Interpreter {
```

Comment:
Likewise, can you write a description for this object (and in general the approach you took that led to this) inside of the pull-request? Will also be useful for @AlexXiong97 who is reviewing this PR, too.